js/app.js

```diff
@@ -65,16 +64,43 @@ import '../css/main.css';
 const createStoreWithMiddleware = applyMiddleware(thunk)(createStore);
 const store = createStoreWithMiddleware(homeReducer);
 
+function checkAuth(nextState, replace) {
```

Comment:
nitpick: rename `replace` to `replaceState` for easier understanding what this function does

Move `checkAuth` function to it's own file in the `utils/` folder please!